### PR TITLE
Fixes #22430: Healthcheck on file descriptor should be max 64000

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/22430

### CODE REVIEW

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala [hunk 1]
```diff
@@ -152,13 +152,16 @@ final class CheckFileDescriptorLimit(val nodeInfoService: NodeInfoService) exten
...
+      val reasonableMaxLimit = 64000
+      val approximatedMinLimit = 100 * nodeCount
+      // 64000 seems to be more than enough even with hundreds of nodes see https://issues.rudder.io/issues/22430
+      val minimalLimit = if (approximatedMinLimit > reasonableMaxLimit) reasonableMaxLimit else approximatedMinLimit
       limit match {
         case limit if limit <= 10_000 =>
           Critical(name, s"Current file descriptor limit is ${limit}. It should be > 10 000.")
-        case limit if limit <= minimalLimit =>
-          Warning(name, s"Current file descriptor limit is ${limit}. It should be > ${minimalLimit} for ${numNode} nodes")
+        case limit if limit <= approximatedMinLimit =>
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala:163]: It should be minimalLimit

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Review by ncharles — APPROVED** [review]

**Comment by ElaadF** [comment]
OK, merging this PR

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala (hunk 1 above)
Comment (ncharles) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/healthcheck/HealthcheckService.scala]: it should have been <

**Pull request merged**